app/src/lib/stores/app-store.ts

```diff
@@ -3839,18 +3849,70 @@ export class AppStore extends TypedBaseStore<IAppState> {
     this.hasUserViewedStash = false
   }
 
-  private async refreshAfterCheckout(repository: Repository, branch: Branch) {
+  private async refreshAfterCheckout(
```

Comment:
We use the git concept of `commitish` elsewhere in the code where a string may be any manner of referencing a commit whether that be a branch name, commit sha, or other ways to denote a commit. 

Could we update this method to take a single parameter of `commitish: string` instead of two  possibly null parameters - update he use cases to send in the narrowed string ref?  Along that same lines, since checking out a branch and checking out a commit share the same progress interface. Could we also update the `ICheckoutProgress` interface in `progress.ts` to be `commitish` instead of target branch and update the doc to reflect that it could a branch name or a commit.

Reply:
Addressed in 10f673f39260178ce94dcfccf2ee86bdb8d61233. Please check.